### PR TITLE
Generate modulation signal from unconnected processor

### PR DESCRIPTION
**Pull request by KaiMikkelsen**

### CODE REVIEW

**Comment by KaiMikkelsen** [comment]
Could be way off on this solution, from the testing I've done it seems to be working as we want but there could be an edge case I haven't thought of.

File: src/processors/chain/ProcessorChain.cpp [hunk 1]
```diff
@@ -175,8 +175,20 @@ void ProcessorChain::processAudio (AudioBuffer<float>& buffer)
             inputBuffer.copyFrom (ch, 0, osBlock.getChannelPointer ((size_t) ch), osNumSamples);
     }
 
-    // run processing chain
+    // process standalone modulation ports
     bool outProcessed = false;
+    for (auto* processor : procs)
+    {
+        auto numImputs = processor->getNumInputConnections();
+
```
Comment (jatinchowdhury18) [inline src/processors/chain/ProcessorChain.cpp:183]:
```suggestion
```

File: src/processors/chain/ProcessorChain.cpp [hunk 2]
```diff
@@ -175,8 +175,20 @@ void ProcessorChain::processAudio (AudioBuffer<float>& buffer)
...
+        auto isModProcessor = processor->getProcessorType() == Modulation;
```
Comment (jatinchowdhury18) [inline src/processors/chain/ProcessorChain.cpp:184]: Is this redundant? Like I'm thinking that if this is false, then `modOutputConnected` would also be false, so maybe we don't need to explicitly check this as well? Or maybe we should just move this check into `BaseProcessor::isOutputModulationPortConnected()`?

**Review by jatinchowdhury18 — APPROVED** [review]
Yeah, this seems to work well. Let's keep testing this after we merge, just to see if there's a way we can figure out how to break it... but I think this is the solution! Nice work :)

File: src/processors/chain/ProcessorChain.cpp (hunk 2 above)
Comment (KaiMikkelsen) [reply src/processors/chain/ProcessorChain.cpp]: Yeah it seems like we don't need it after doing some testing, good call. The only thing I can think of is checking if it's a modulation processor before looping through the connections to save doing that if it's not a modulation processor?

File: src/processors/BaseProcessor.cpp [hunk 3]
```diff
@@ -286,3 +286,19 @@ bool BaseProcessor::isOutputModulationPort (int portIndex)
 {
     return outputModulationPorts.contains (portIndex);
 }
+
+bool BaseProcessor::isOutputModulationPortConnected()
+{
+    if (getProcessorType() != Modulation)
+        return false;
```
Comment (jatinchowdhury18) [inline src/processors/BaseProcessor.cpp:293]: Yeah i think this is good.

**Pull request merged**